Validate compatibility with latest Error Prone release

Stephan202 opened the pull request:
See discussion in #1646. I opted not to bother with testing against a "locally" built HEAD release of Error Prone, as that would at most speed up incompatibility identification by 24 hours, while potentially causing weeks/months of build failures between the time an incompatibility is introduced and a subsequent release.

Suggested commit message:
```
Validate compatibility with latest Error Prone release (#1651)

Every day and after each merged change we now run a GitHub Actions
workflow that builds the project against the latest Error Prone release,
as determined by Maven. This should pro-actively inform us of
incompatibilities that demand a new release.

While there:
- Drop a redundant permission declaration from the SonarCloud workflow.
- Update `run-full-build.sh` to parallelize the build.
```

Stephan202 commented:
New check fails as expected. Build should pass after #1649.

rickie reviewed (commented):
Didn't review in depth, but very nice set up @Stephan202 !

Stephan202 commented:
Rebased and added a commit to track a [nice suggestion](https://github.com/PicnicSupermarket/error-prone-support/issues/1646#issuecomment-2816824543) by @ben-manes.

The build now passes, so dropped the test logic. That means the PR is ready for review.

mohamedsamehsalah commented on run-full-build.sh:
```diff
@@ -11,9 +11,11 @@ set -e -u -o pipefail
 settings="$(dirname "${0}")/settings.xml"
 
 mvn clean install \
+  -T1C \
```
⚙️

mohamedsamehsalah reviewed (approved):
Nice 🌞
